CodeEditModules/Modules/CodeFile/src/TextView/CodeEditorTextView.swift

```diff
@@ -7,8 +7,8 @@
 
 import Cocoa
 
-class CodeEditorTextView: NSTextView {
-    private let tabNumber = 4
+public class CodeEditorTextView: NSTextView {
+    public static var tabWidth = UserDefaults.standard.value(forKey: "defaultTabWidth") as? Int ?? 4
```

Comment:
Use `@AppStorage` instead

```swift
@AppStorage(EditorTabWidth.storageKey) var tabWidth: Int = EditorTabWidth.default
```

Reply:
@lukepistrol I'd love to do this, but I can't access `EditorTabWidth` within the `CodeFile` module, the `Settings > Models` folder lies within the app project, I don't think importing the app into a module is a good idea (or even possible). I would need to move the tab width enum to the module then, is that fine with you? Should I even extract the entire "Settings" folder into a module (if possible easily)?

Reply:
Also, can I please put `@AppStorage(EditorTabWidth.storageKey)` into its own line? The code is so hard to skim through with these `var ...` starting so much to the right, I think all attributes should always be at the start of the line so the code is easier to scan. I don't know why Apple is defaulting to putting it all in one line in their code samples, but I think it's less readable ...

Reply:
Moving it to the `CodeFile` module is good for now. @MarcoCarnevali is already working on moving everything to its own module but this takes some time. For now put it into the `CodeFile` module and we will move it to the dedicated `Settings` module once the migration is done!

Reply:
> Also, can I please put `@AppStorage(EditorTabWidth.storageKey)` into its own line? The code is so hard to skim through with these `var ...` starting so much to the right, I think all attributes should always be at the start of the line so the code is easier to scan. I don't know why Apple is defaulting to putting it all in one line in their code samples, but I think it's less readable ...

Yeah go ahead but please make sure to change every `@AppStorage` statement so it is the same everywhere

Reply:
I just added the `EditorTabWidth` to the module instead of the app, this way it worked. See cd76e3460356333e35bfb4d48f76025019307198.